pyproject.toml

```diff
@@ -0,0 +1,10 @@
+[tool.black]
+line-length = 100
```

Comment:
Could we add the option to not touch the string quotes (whatever the equivalent of `let g:black_skip_string_normalization=1` is). This reduces unneeded disruption.

Reply:
You mean ' to "?
But this I think we want for new code (as @RoyStegeman said, this is just the file that black and isort use for their config options when you run)

Reply:
I think it is likely enough that this gets executed in old files (where `git add -uip` would give you a point of pointless changes to sort trough) and also I don't particularly love the setting in new code, even if I'll use it if it's set already: it is the opposite choice of the Python repr, people do like to take issue with that https://github.com/grantjenks/blue, and personally I was used to https://stackoverflow.com/a/56190, from before people decided that having two string quotes was intolerable.

Reply:
We started using `black` upon your suggestion.

That said, I don't have any strong opinion about style as long as it is consistent. Feel free to add any options to this `.toml` file.

Honestly, what would be great is to arrive to a set of options we are happy with and then run it for the entire codebase, skip the particular commit in the blame and continue from there.

Reply:
Do those preferences really affect your efficiency though? I don't mean this to argue, but genuinely wondering.

If I had to do things manually I might decide to make it a bit more fancy as well (indeed for example https://stackoverflow.com/a/56190), but 
1. that doesn't really work with multiple people, unless we're going to put way more emphasis on formatting guidlines, so we'll have to end up accepting different formats anyway (if we don't use a tool, that is)
2. More than getting to choose my own formatting I like not having to spend mental energy on it at all (within reason, e.g. `pdfbases.py` would probably require some care).

Reply:
> Honestly, what would be great is to arrive to a set of options we are happy with and then run it for the entire codebase, skip the particular commit in the blame and continue from there.

Is this reasonable considering the different PRs that are open at any given moment?

Reply:
I was expressing a desire but I realise it probably won't happen.

Reply:
When I am working on a personal project I enable the option to not touch the string marks. When I am working on a project that does change the quotes, then I do that. But that is not the main point I was making.

The argument is that for this project, in this context, it is better to have the option because it reduces the diffs when reformatting stuff, accidentally or otherwise, and because enabling the option is arguably of little value overall.

IMO the way to deal with string quotes in this project is not to add review comments expressing the desire to Marie Kondo the world in regards to string quotes (which is what has happened so far without particular issues).

Reply:
While I don't necessarily agree I would be happy to add `skip-string-normalization = 1` (assuming that's correct) to the `.toml` file. The main benefit of black for me would be a universal way of enforcing line-length and indentation anyway,  the quotations don't bother me as much. 

Having said that - consistent quotations do spark joy :)

Reply:
To come back to this, would both of  you be ok with `skip-string-normalization = 1`? It still  won't enforce anything but with this at least there would be consistency in output if we ask people to run black.